src/validate_pyproject/plugins/setuptools.schema.json

```diff
@@ -226,9 +226,16 @@
           }
         },
         "readme": {
+          "type": "object",
           "anyOf": [
             {"$ref": "#/definitions/file-directive"},
-            {"properties": {"content-type": {"type": "string"}}}
+            {
+              "type": "object",
+              "properties": {
+                "content-type": {"type": "string"},
+                "file": { "$ref": "#/definitions/file-directive/properties/file" }
+              },
+              "additionalProperties": false}
           ],
           "required": ["file"]
```

Comment:
FYI, this could be moved inside the second `anyOf` item, however I liked the error message better this way, so unless it's required to make the Schema compliant to something, which it doesn't seem to be, I think putting it here is fine.